### PR TITLE
DPP 0.15

### PR DESCRIPTION
**Pull request by HashEngineering**
Upgrading to DPP 0.15:  https://blog.dash.org/release-announcement-dash-platform-v0-15-on-evonet-88131f9c477b

These are the two changes that are related to DPP 0.15 according to the Blog:
**Document Binary Properties**: There are many instances where developers need to store binary data in their applications. Previously, they had to encode and store them as strings, which required additional space in platform state and increased state transition fees. Now, developers have the ability to define those binary fields directly in a data contract by specifying the `contentEncoding` keyword. This allows developers to use native binary types (e.g., Buffer, ByteArray) to store their data.

**What changes does this PR make**:  None.  Binary fields and the conversion of ByteArray to and from  CBOR are already supported by this library.

**Protocol Versioning**: As Dash Platform evolves over time, it is likely that breaking changes will be introduced on the protocol level (i.e., hard forks). This would lead to various incompatibility issues between previously created data and new software versions, as well as different versions of clients and nodes participating in the same network. Previously, DCG developers would handle these situations by wiping data on Evonet and forcing developers to update platform software with each new release. While this approach is fine for early testing, it is not appropriate for testnet and mainnet. To handle different versions of data, nodes, and clients all working in coordination, DCG developers introduced versioning that allows for the smooth rollout of breaking changes without affecting user experience.

**What changes does this PR make**:  DataContract now has a protocol version field.  Unit tests were updated to account for this field.

Other changes:

1. Removed obsolete tests and related data files

### CODE REVIEW

File: src/main/kotlin/org/dashevo/dpp/contract/DataContract.kt [hunk 1]
```diff
@@ -12,18 +12,23 @@ import org.dashevo.dpp.errors.InvalidDocumentTypeError
...
+        const val PROTOCOL_VERSION: Int = 0
     }
 
     var entropy: String? = null
 
     constructor(rawContract: MutableMap<String, Any?>) : this(rawContract["\$id"] as String,
             rawContract["ownerId"] as String,
+            if (rawContract.containsKey("protocolVersion"))
+                rawContract["protocolVersion"] as Int
+            else 0,
```
Comment (sambarboza) [inline src/main/kotlin/org/dashevo/dpp/contract/DataContract.kt:31]: In this case `0` means the same as `PROTOCOL_VERSION` or we want to set the value strictly to `0`. If it means the latest `PROTOCOL_VERSION` as default, we should probably use the cosnt from above.

**Review by sambarboza — APPROVED** [review]

File: src/main/kotlin/org/dashevo/dpp/contract/DataContract.kt (hunk 1 above)
Comment (HashEngineering) [reply src/main/kotlin/org/dashevo/dpp/contract/DataContract.kt:31]: Good point.  This will be a bug later and is not the right way to handle this.

Comment (HashEngineering) [reply src/main/kotlin/org/dashevo/dpp/contract/DataContract.kt]: https://github.com/dashevo/android-dpp/pull/9/commits/e5fb399bc3a9284e2417521069104518d63faa5b - fixed here.

Comment (HashEngineering) [reply src/main/kotlin/org/dashevo/dpp/contract/DataContract.kt]: https://github.com/dashevo/android-dpp/pull/9/commits/e5fb399bc3a9284e2417521069104518d63faa5b - fixed here

**Pull request merged**